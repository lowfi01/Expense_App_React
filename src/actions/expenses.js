import uuid from 'uuid';

// Add Expense
export const addExpense = (
  {
      description = '',
      note = '',
      amount = 0,
      createdAt = 0
  } = {}
) => ({
  type: 'ADD_EXPENSE',
  expense: {
    id: uuid(),
    description,
    note,
    amount,
    createdAt
  }
})


// Remove Expense
export const removeExpense = ({id}) => ({
  type: 'REMOVE_EXPENSE',
  id
})


// Edit Expense
export const editExpense = (id, updates ) => ({
  type: 'EDIT_EXPENSE',
  id,
  updates
});
