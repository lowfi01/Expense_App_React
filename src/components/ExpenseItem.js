import React from 'react';
import { Link } from 'react-router-dom';

export const ExpenseItem = ({id, description, amount, createdAt, dispatch}) => (
    <div>
      <Link to={`/edit/${id}`}>
        <h3>{description}</h3>
      </Link>
      <p>{amount}-{createdAt}</p>

    </div>
  )

export default ExpenseItem;