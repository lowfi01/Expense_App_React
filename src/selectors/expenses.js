/* eslint-disable */



import moment from 'moment';

export default (expenses, {text, sortBy, startDate, endDate}) =>
  // remember the combineReducers, uses key value pairs for the object it makes
  // we have these two reducers in our store
  // {
  //  filters: filterReducer,
  //  expenses: expensesReducer
  // }

    expenses.filter((expense) => {
      // const startDateMatch = typeof startDate !== 'number' || expense.createdAt >= startDate ;
      // const endDateMatch = typeof endDate !== 'number' || expense.createdAt <= endDate ;

      const createdAtMoment = moment(expense.createdAt);
      const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
      const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
      const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());

      return startDateMatch && endDateMatch && textMatch;
   }).sort((a, b) => { // eslint-disable-line

    if(sortBy === 'date') {
      return  a.createdAt < b.createdAt ? 1 : -1;x
     } else if (sortBy === 'amount') {
      return a.amount < b.amount ? 1 : -1;
     }

    })  // sort by date or amount, depending on filter.sortBy definition


// export default getVisibleExpenses;