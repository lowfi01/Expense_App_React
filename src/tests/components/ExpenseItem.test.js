import React from 'react';
import { shallow } from 'enzyme';
import ExpenseItem from '../../components/ExpenseItem';
import expenses from '../fixtures/expenses';

test('should render a expense item with expense data', () => {
  const wrapper = shallow(<ExpenseItem {...expenses[0]}/>);
  expect(wrapper).toMatchSnapshot();
})
