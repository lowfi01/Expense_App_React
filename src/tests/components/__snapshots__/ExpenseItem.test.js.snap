// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a expense item with expense data 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    199
    -
    0
  </p>
</div>
`;
